clients/java/src/main/java/io/camunda/zeebe/client/impl/CamundaClientImpl.java

```diff
@@ -0,0 +1,523 @@
+import io.camunda.zeebe.client.api.response.ActivatedJob;
+import io.camunda.zeebe.client.api.search.ProcessInstanceQuery;
+import io.camunda.zeebe.client.api.worker.JobClient;
+import io.camunda.zeebe.client.api.worker.JobWorkerBuilderStep1;
+import io.camunda.zeebe.client.impl.command.AssignUserTaskCommandImpl;
+import io.camunda.zeebe.client.impl.command.BroadcastSignalCommandImpl;
+import io.camunda.zeebe.client.impl.command.CancelProcessInstanceCommandImpl;
+import io.camunda.zeebe.client.impl.command.CompleteUserTaskCommandImpl;
+import io.camunda.zeebe.client.impl.command.CreateProcessInstanceCommandImpl;
+import io.camunda.zeebe.client.impl.command.DeleteResourceCommandImpl;
+import io.camunda.zeebe.client.impl.command.DeployProcessCommandImpl;
+import io.camunda.zeebe.client.impl.command.DeployResourceCommandImpl;
+import io.camunda.zeebe.client.impl.command.EvaluateDecisionCommandImpl;
+import io.camunda.zeebe.client.impl.command.JobUpdateRetriesCommandImpl;
+import io.camunda.zeebe.client.impl.command.JobUpdateTimeoutCommandImpl;
+import io.camunda.zeebe.client.impl.command.MigrateProcessInstanceCommandImpl;
+import io.camunda.zeebe.client.impl.command.ModifyProcessInstanceCommandImpl;
+import io.camunda.zeebe.client.impl.command.PublishMessageCommandImpl;
+import io.camunda.zeebe.client.impl.command.ResolveIncidentCommandImpl;
+import io.camunda.zeebe.client.impl.command.SetVariablesCommandImpl;
+import io.camunda.zeebe.client.impl.command.StreamJobsCommandImpl;
+import io.camunda.zeebe.client.impl.command.TopologyRequestImpl;
+import io.camunda.zeebe.client.impl.command.UnassignUserTaskCommandImpl;
+import io.camunda.zeebe.client.impl.command.UpdateUserTaskCommandImpl;
+import io.camunda.zeebe.client.impl.http.HttpClient;
+import io.camunda.zeebe.client.impl.http.HttpClientFactory;
+import io.camunda.zeebe.client.impl.search.ProcessInstanceQueryImpl;
+import io.camunda.zeebe.client.impl.util.ExecutorResource;
+import io.camunda.zeebe.client.impl.util.VersionUtil;
+import io.camunda.zeebe.client.impl.worker.JobClientImpl;
+import io.camunda.zeebe.client.impl.worker.JobWorkerBuilderImpl;
+import io.camunda.zeebe.gateway.protocol.GatewayGrpc;
+import io.camunda.zeebe.gateway.protocol.GatewayGrpc.GatewayStub;
+import io.grpc.CallCredentials;
+import io.grpc.ClientInterceptor;
+import io.grpc.ManagedChannel;
+import io.grpc.netty.GrpcSslContexts;
+import io.grpc.netty.NettyChannelBuilder;
+import io.netty.handler.ssl.SslContext;
+import java.io.Closeable;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public final class CamundaClientImpl implements CamundaClient {
```

Comment:
❓ For many other classes, the Camunda classes extend the existing Zeebe classes. Is this not possible here?

Reply:
Nope